crates/zkvm/adapters/native/src/host.rs

```diff
@@ -15,36 +15,33 @@ pub struct NativeHost {
 
 impl ZkVmHost for NativeHost {
     type Input<'a> = NativeMachineInputBuilder;
+    type ZkVmProofReceipt = NativeProofReceipt;
 
-    fn prove<'a>(
+    fn prove_inner<'a>(
         &self,
         native_machine: NativeMachine,
         _proof_type: ProofType,
-    ) -> ZkVmResult<ProofReceipt> {
+    ) -> ZkVmResult<NativeProofReceipt> {
```

Comment:
Putting the `ZkVm` prefix on this assoc type feels redundant.  And then we could just have the return type still be `Self::ProofReceipt` instead.

Reply:
We already have a `ProofReceipt` struct defined. So added `ZkVm` prefix for the associated types.